Adyen/Core/APIClient/APIContext.swift

```diff
@@ -0,0 +1,50 @@
+//
+// Copyright (c) 2021 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Foundation
+
+/// :nodoc:
+/// An object that needs an API context to retrieve internal resources
+public protocol APIContextAware: AnyObject {
+    
+    /// :nodoc:
+    /// The API context
+    var apiContext: APIContext { get }
+    
+}
+
+/// Struct that defines API context for retrieving internal resources.
+public struct APIContext {
```

Comment:
without protocol this struct looks "too concrete".. 😅
But it is not like we expecting another remote service in foreseeable future.